python/tests/conftest.py

```diff
@@ -148,8 +148,17 @@ def _clean_nessie_session_marker(request: "FixtureRequest", record_mode: str) ->
     if record_mode is None or record_mode == "none":
         return
 
-    # Clean the server state for tests that have the @pytest.mark.clean_nessie_session annotation
-    if request.node.get_closest_marker("vcr"):
+    vcr_marker = request.node.get_closest_marker("vcr")
+
+    # Clean the server state for tests that have the @pytest.mark.vcr annotation
+    if vcr_marker:
+        if "record_mode" in vcr_marker.kwargs:
+            # if we have record_mode set in @pytest.mark.vcr annotation, we test that first
+            test_record_mode = vcr_marker.kwargs["record_mode"]
+
+            if test_record_mode is None or test_record_mode == "none":
```

Comment:
I'd treat `None` as an error in this case. Setting `record_mode` without a value does not look intuitive for _disabling_ the recording.

Reply:
Fixed